File: src/main/kotlin/fr/jhelp/kotlinLightSamples/EqualsTest.kt
package fr.jhelp.kotlinLightSamples

class EqualsTest
{
    val info: String

    constructor(info: String)
    {
        this.info = info
    }

    override fun equals(other: Any?): Boolean
    {
        if (null == other || (other !is EqualsTest))
        {
            return false
        }

        val otherEquals = other!! as EqualsTest
        return this.info == otherEquals.info
    }
}

File: src/main/swift/Tests/swiftTests/src/jhelp/kotlinLightSamples/EqualsTest.swift

public class EqualsTest : Equatable {

     static public func == (lhs: EqualsTest, rhs: EqualsTest) -> Bool
     {
          return lhs.equals(rhs)
     }


    let info: String

    public init(_ info: String){
        self.info = info
    }

    public func equals(_ other: Any?) -> Bool
    {
        if nil == other || !(other is EqualsTest)
        {
            return false
        }

        let otherEquals = other as! EqualsTest
        return self.info == otherEquals.info
    }
}
